src/middlewared/middlewared/plugins/zfs_/pool_wait.py

```diff
@@ -0,0 +1,30 @@
+import libzfs
+
+from middlewared.schema import accepts, Dict, Str
+from middlewared.service import CallError, Service
+
+
+POOL_ACTIVITY_TYPES = [a.name for a in libzfs.ZpoolWaitActivity]
+
+
+class ZFSPoolService(Service):
+
+    class Config:
+        namespace = 'zfs.pool'
+        private = True
+        process_pool = True
+
+    @accepts(
+        Str('pool_name'),
+        Dict(
+            'options',
+            Str('activity_type', enum=POOL_ACTIVITY_TYPES, required=True),
+        )
+    )
+    def wait(self, pool_name, options):
```

Comment:
Since this is public, do we not need a `returns()` ?

Reply:
Nope, the service is private ;)

Reply:
Ahh right